src/game/game.cpp

```diff
@@ -10532,3 +10556,25 @@ std::vector<Achievement> Game::getPublicAchievements() {
 std::map<uint16_t, Achievement> Game::getAchievements() {
 	return m_achievements;
 }
+
+void Game::getCyclopediaStatistics() {
+	g_logger().info("Loaded {} badges from Badge System", m_badges.size());
+	// todo in title system: g_logger().info("Loaded {} titles from Title system", m_titles.size());
+}
+
+std::unordered_set<Badge> Game::getBadges() {
+	return m_badges;
+}
+
+Badge Game::getBadgeByIdOrName(uint8_t id, const std::string &name /*= ""*/) {
```

Comment:
I recommend separating this function into two distinct functions, getBadgeById and getBadgeByName. This change would enhance clarity by assigning a single, focused responsibility to each function. It would also improve readability, as developers can quickly understand what each function does without having to parse through conditional logic that handles two different types of inputs. Furthermore, maintenance would be simplified since any future changes or debugging efforts could be localized to the specific function relevant to the issue at hand. Overall, this separation adheres to the Single Responsibility Principle and can lead to cleaner and more maintainable code.